File: Backend/src/main/java/com/example/easynotes/model/DiaChi.java
package com.example.easynotes.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import javax.validation.constraints.NotBlank;
import java.io.Serializable;


@Embeddable
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DiaChi implements Serializable {
    @NotBlank
    @Column(name = "soNha")
    private String soNha;

    @NotBlank
    private String phuong;
    @NotBlank
    private String quan;
    @NotBlank
    private String tinh;
}

File: App/go_quick_app/lib/models/dia_chi.dart
class DiaChi {
  DiaChi({
    this.soNha,
    this.phuong,
    this.quan,
    this.tinh,
  });

  String? soNha;
  String? phuong;
  String? quan;
  String? tinh;

  factory DiaChi.fromJson(Map<String, dynamic> json) => DiaChi(
        soNha: json["soNha"],
        phuong: json["phuong"],
        quan: json["quan"],
        tinh: json["tinh"],
      );

  Map<String, dynamic> toJson() => {
        "soNha": soNha,
        "phuong": phuong,
        "quan": quan,
        "tinh": tinh,
      };
}
